AssemblerLib/VectorMatrixAssembler.h

```diff
@@ -38,6 +38,14 @@ class VectorMatrixAssembler
 
     ~VectorMatrixAssembler() {}
 
+    void setX(GLOBAL_VECTOR_ const * x, GLOBAL_VECTOR_ const * x_prev_ts)
+    {
+        assert((!x == !x_prev_ts) && "either no or both inputs have to be set");
+        assert((!x) || x->size() == x_prev_ts->size());
+        _x = x;
+        _x_prev_ts = x_prev_ts;
+    }
```

Comment:
Passing the solution `x` to assembler is required for the non-linear assembly; OK.
How is the `x_prev_ts` used?